cuqi/experimental/mcmc/_sampler.py

```diff
@@ -229,6 +230,9 @@ def sample(self, Ns, batch_size=0, sample_path='./CUQI_samples/') -> 'Sampler':
             self._acc.append(acc)
             self._samples.append(self.current_point)
 
+            # display acc rate at progress bar
+            pbar.set_postfix_str(f"acc rate: {np.mean(self._acc):.2%}")
```

Comment:
I have a comment here and I am not sure what is the best decision. Maybe @nabriis can also weigh in. Consider for example the code

```python
import cuqi
import numpy as np
np.random.seed(10)

uniform = cuqi.distribution.Uniform(np.array([0.0]), np.array([1.0]))
sampler = cuqi.experimental.mcmc.NUTS(uniform, initial_point=np.array([0.1]))
sampler.sample(Ns=10)
sampler.sample(Ns=10)
sampler.sample(Ns=10)
sampler.sample(Ns=10)
sampler.sample(Ns=10)
sampler.sample(Ns=2)
```

When reporting the acceptance rate, do we want to report the acceptance rate of the Ns samples or for all the samples from before? The code behavior here seems to be the latter (all the samples, including ones obtained in previous `sample` calls).  

To make it only for the Ns samples maybe we can do some updates like
```python
for sample_idx in pbar:
   # ..
   # code ..
   # .. 
    pbar.set_postfix_str(f"acc rate: {np.mean(self._acc[-1-sample_idx:]):.2%}")

```

This also applies to the warmup method

Reply:
Done as suggested

Reply:
Seems reasonable to me.